needle.pm

```diff
@@ -327,6 +327,7 @@ sub default_needles_dir {
 sub init {
     $needles_dir = ($bmwqemu::vars{NEEDLES_DIR} // default_needles_dir);
     $needles_dir = File::Spec->catdir($bmwqemu::vars{CASEDIR}, $needles_dir) unless -d $needles_dir;
+    $needles_dir = "$bmwqemu::vars{INIT_PRODUCTDIR}/needles"                 unless -d $needles_dir;
```

Comment:
hm, I am thinking that this isn't a very clear design because the variable INIT_PRODUCTDIR has no meaning within the scope of os-autoinst and having an incorrect PRODUCTDIR specified for os-autoinst is also not something that makes much sense. Keep in mind that isotovideo as standalone tool cares only about a single test with no history of "previous jobs" or anything. As the "check for a needle directory" is actually only a check if a directory with name "needles" exists I think here it would be better if you move the corresponding logic to openQA code. So it would be like the openQA worker code tries to "repair" the variable PRODUCTDIR in https://github.com/os-autoinst/openQA/pull/3578/files#diff-53098d895ef28f5c7db596e2dc505d7c8ee24b6a9a4cca23865fb225e10cd9fcR332 if otherwise no needle can be found.

Reply:
In general, we don't need `INIT_PRODUCTDIR`, but when the `PRODUCTDIR` is a relative path (for example the job is trigger by `openqa-clone-custom-git-refspec`), we need this to find the `needles_dir`. 
 
>   I think here it would be better if you move the corresponding logic to openQA code.
if we move this logic to openQA code, when users specify `NEEDLES_DIR` as a GitHub or GitLab path,  it will not work. Because the git check out happens in `checkout_git_repo_and_branch` on https://github.com/os-autoinst/os-autoinst/blob/master/OpenQA/Isotovideo/Utils.pm#L50